sympy/solvers/bivariate.py

```diff
@@ -302,6 +305,9 @@ def _solve_lambert(f, symbol, gens):
         raise NotImplementedError('%s does not appear to have a solution in '
             'terms of LambertW' % f)
 
+    soln = [s for s in soln if
+              checksol(f, {symbol: s}, **flags) is not False]
```

Comment:
Is `checksol` really needed? Can you mention some example which otherwise would add incorrect solutions?

Reply:
`flags` is not required as an argument.

Reply:
Yes In this following case it is required in which we only use `_solve_lambert`.
```
p = symbols('p', positive=True)
eq = 4*2**(2*p + 3) - 2*p - 3
_solve_lambert(eq, p, _filtered_gens(Poly(eq), p))
[-3/2 - LambertW(-4*log(2))/(2*log(2))] # if checksol included for checking correct solutions
[-3/2 - LambertW(-4*log(2))/(2*log(2)), -3/2 - LambertW(4*log(2))/(2*log(2))] # if checksol don't included.
```